app/src/main/res/layout/fragment_subscriptions.xml

```diff
@@ -1,69 +1,74 @@
+    android:orientation="vertical">
 
     <androidx.appcompat.widget.Toolbar
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:minHeight="?attr/actionBarSize"
-            android:theme="?attr/actionBarTheme"
-            android:layout_alignParentTop="true"
-            app:title="@string/subscriptions_label"
-            app:navigationIcon="?homeAsUpIndicator"
-            android:id="@+id/toolbar"/>
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:minHeight="?attr/actionBarSize"
+        android:theme="?attr/actionBarTheme"
+        android:layout_alignParentTop="true"
+        app:title="@string/subscriptions_label"
+        app:navigationIcon="?homeAsUpIndicator" />
 
     <TextView
-            android:id="@+id/feeds_filtered_message"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_below="@id/toolbar"
-            android:background="?android:attr/selectableItemBackground"
-            android:gravity="start"
-            android:paddingStart="8dp"
-            android:paddingTop="4dp"
-            android:paddingEnd="8dp"
-            android:paddingBottom="8dp"
-            android:textColor="?android:attr/textColorSecondary"
-            android:textSize="@dimen/text_size_small" />
+        android:id="@+id/feeds_filtered_message"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/toolbar"
+        android:background="?android:attr/selectableItemBackground"
+        android:gravity="start"
+        android:paddingStart="8dp"
+        android:paddingTop="4dp"
+        android:paddingEnd="8dp"
+        android:paddingBottom="8dp"
+        android:textColor="?android:attr/textColorSecondary"
+        android:textSize="@dimen/text_size_small" />
 
     <androidx.swiperefreshlayout.widget.SwipeRefreshLayout
         android:id="@+id/swipeRefresh"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:layout_below="@id/feeds_filtered_message">
 
-        <GridView
-                android:id="@+id/subscriptions_grid"
-                android:layout_width="match_parent"
-                android:numColumns="3"
-                android:horizontalSpacing="2dp"
-                android:verticalSpacing="2dp"
-                android:layout_height="match_parent"
-                android:layout_gravity="center_horizontal"
-                android:paddingBottom="88dp"
-                android:clipToPadding="false"/>
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/subscriptions_grid"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:numColumns="3"
```

Comment:
Is this an attribute that RecyclerViews support? Same for the spacing. I think they can all be removed

Reply:
Well since the spacing is set dynamically spacing attribute isn't needed. Same with numColumns. Removed both.